### ISSUE TITLE
Filter convolution performance

### ISSUE THREAD

**Issue by andcastillo** [image-js/core#110]
@lpatiny  and @targos : Do you have any limit for the size of the images or to the filters that you are going to use? Because I did a bench mark about the performance of your Convolution function vs the FT-Convolution ([https://github.com/mljs/fft/commit/6c47f131898e215360707127331603698e2ebd41]) and here are the results(Time is in millisecond and only one run): 

Rows: Matrix size h/w 256, 512, 1024, 2048
Columns: Filter size h/w 5, 11, 19, 27, 33, 37

Normal 
[ [ 11, 38, 99, 180, 264, 314 ],
  [ 37, 172, 420, 835, 1338, 1522 ],
  [ 155, 651, 1754, 3445, 5318, 6683 ],
  [ 633, 2556, 7333, 14090, 22420, 28736 ] ]

FFT
[ [ 55, 19, 14, 14, 13, 14 ],
  [ 103, 82, 90, 93, 99, 92 ],
  [ 383, 356, 384, 394, 360, 389 ],
  [ 1616, 1604, 1646, 1621, 1602, 1653 ] ]

Basically for the small filters(First column) it is better to use the normal convolution, but for all the other sizes in the bench mark, to convolute in the Fourier space is much more faster (28 sec vs 1.6 secs) for a 2Kx2K image with a  37x37 filter.

**Comment by andcastillo** [comment]
Another way to see the results is that the convolution only scales with the size of the image, and it keeps constant the time for any size of the kernel( Obviously smaller than the image)
